src/test/java/org/broadinstitute/hellbender/utils/recalibration/covariates/ContextCovariateUnitTest.java

```diff
@@ -46,12 +48,42 @@ public void testSimpleContexts() {
         }
     }
 
-    public static void verifyCovariateArray(int[][] values, int contextSize, GATKRead read, Covariate contextCovariate, final byte lowQualTail) {
+
+    private static void verifyCovariateArray(final int[][] values, final int contextSize, final GATKRead read, final Covariate contextCovariate, final byte lowQualTail) {
         for (int i = 0; i < values.length; i++) {
             Assert.assertEquals(contextCovariate.formatKey(values[i][0]), expectedContext(read, i, contextSize, lowQualTail), "offset " + i);
         }
     }
 
+    @DataProvider
+    Iterator<Object[]> AnnoyingReads() {
+        final List<Object[]> tests = new ArrayList<>();
+        for (final byte base : "NBACGTUMLA".getBytes()) {
+            for (int i = 0; i < 10; i++) {
+                tests.add(new Object[]{base, i});
+            }
+        }
+        return tests.iterator();
+    }
+
+    @Test(dataProvider = "AnnoyingReads")
+    public void testContextsAnnoyingReads(final byte base, final int i) {
+        final Random rnd = Utils.getRandomGenerator();
+        final SAMFileHeader header = ArtificialReadUtils.createArtificialSamHeader();
+        final GATKRead read = ArtificialReadUtils.createRandomRead(header, 10,false);
+        read.getBasesNoCopy()[i] = base;
```

Comment:
I think it's worth having a couple of test cases that have more than one "annoying base." Won't the code you patched behave differently in the case of 1 non-ACGT base vs. the case of 2 non-ACGT bases?

Reply:
ok.